mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOpsTypes.h

```diff
@@ -332,6 +346,16 @@ class AccessorType
   llvm::ArrayRef<mlir::Type> getBody() const;
 };
 
+class AccessorCommonType
```

Comment:
I think the ` AccessorType` should be use 
 `mlir::sycl::SYCLInheritanceTypeInterface<mlir::sycl::AccessorCommonType>::Trait>` similar to the patter for example in `RangeType`